### ISSUE TITLE
Question to CascadePSP

### ISSUE THREAD

**Issue by ChristophReich1996** [frank-xwang/UnSAM#9]
Hi, I have seen you are using CascadePSP for refining the segmentation masks; however, CascadePSP is not unsupervised but trained using GT masks (please correct me if I'm wrong). Have you re-trained CascadePSP in an unsupervised setting, or are you just using the official supervised checkpoint? Thanks for the help, Christoph.

**Comment by frank-xwang** [comment]
Hi, this is a nice question! During the pseudo mask generation process, we follow the previous works in unsupervised image segmentation (CutLER [CVPR 2023] and SOHES [ICLR 2024]), and use CascadePSP (used by SOHES) or CRF (used by CutLER) to refine pseudo masks. The results of these two mask refinement methods are presented in Table 3. We chose CascadePSP (the official checkpoint from its GitHub) as the mask refinement method for two primary reasons: 1) The primary advantage of CascadePSP is not actually in terms of model performance, but rather in terms of inference speed. In our local small scale experiments, we discovered that the performance difference (in terms of AR) between training UnSAM with CascadePSP-refined pseudo-masks and training UnSAM with CRF-refined masks is only about 2-3%, but the speed difference in refining pseudo-masks is about 5-10 times. Because we only have limited computing resources, refining pseudo-masks using CascadePSP took us ~2-3 weeks, and we may not have enough resources to run CRF on the training data. 2) To ensure a fair comparison with the previous state-of-the-art SOHES, we use the same mask refinement model as SOHES.

To fully answer your question, how can we speed up the overall process for CRF-based mask refinement? I believe we can simply train a CascadePSP using the "ground-truth" generated by CRF and train a CascadePSP model from scratch to support "fully-unsupervised" mask refinement with a short inference time. Additionally, I would like to emphasize that the object discovery process that can construct the hierarchical structure of visual scenes is entirely unsupervised.

Hope it answers your questions! Thanks!
